controllers/civocluster_controller.go

```diff
@@ -103,6 +104,13 @@ func (r *CivoClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 
 // TODO Cluster deletion
 func (r *CivoClusterReconciler) delete(ctx context.Context, logger logr.Logger, civoCluster *infrastructurev1beta1.CivoCluster) (ctrl.Result, error) {
+	controllerutil.RemoveFinalizer(civoCluster, civoClusterFinalizer)
```

Comment:
Let's move it to the end of the function) So that finalizer would be removed only at the last step(after cluster is deleted for sure)

Reply:
sure